src/Libraries/CoreNodes/Sorting.cs

```diff
@@ -55,10 +55,18 @@ public static class Sorting
             [ArbitraryDimensionArrayImport] IList list,
             [ArbitraryDimensionArrayImport] IList keys)
         {
-            var sortedPairs = list.Cast<object>()
-                    .Zip(keys.Cast<IComparable>(), (item, key) => new { item, key })
-                    .OrderBy(x => x.key)
-                    .ToList();
+            var pairs = list.Cast<object>()
+                    .Zip(keys.Cast<object>(), (item, key) => new { item, key });
+
+            var numberKeyPairs = pairs.Where(pair => pair.key is double || pair.key is int);
+            var stringKeyPairs = pairs.Except(numberKeyPairs);
```

Comment:
I don't think this should be called stringkeypairs - as it's really ``keys that are not numbers``